src/Utils/PromptNormalizer.php

```diff
@@ -0,0 +1,112 @@
+<?php
+
+declare(strict_types=1);
+
+namespace WordPress\AiClient\Utils;
+
+use WordPress\AiClient\Messages\DTO\Message;
+use WordPress\AiClient\Messages\DTO\MessagePart;
+use WordPress\AiClient\Messages\DTO\UserMessage;
+
+/**
+ * Utility class for normalizing various prompt formats into standardized Message arrays.
+ *
+ * @since n.e.x.t
+ */
+class PromptNormalizer
+{
+    /**
+     * Normalizes various prompt formats into a standardized Message array.
+     *
+     * @since n.e.x.t
+     *
+     * @param string|string[]|MessagePart|MessagePart[]|Message|Message[] $prompt The prompt content in various formats.
+     * @return list<Message> Array of Message objects.
+     *
+     * @throws \InvalidArgumentException If the prompt format is invalid.
+     */
+    public static function normalize($prompt): array
```

Comment:
I think we can actually simplify some of this while making it a bit more flexible by accepting a list of `string|Message|MessagePart`, looping through the list, and simply resolving to a `Message` on a per item basis.

```suggestion
    /**
     * Normalizes various prompt formats into a standardized Message array.
     *
     * @since n.e.x.t
     *
     * @param string|MessagePart||Message|list<string|MessagePart|Message> $prompt The prompt content in various formats.
     * @return list<Message> Array of Message objects.
     *
     * @throws \InvalidArgumentException If the prompt format is invalid.
     */
    public static function normalize($prompt): array
```

You can even do this at the beginning to normalize:
```php
if (!is_array($prompt)) {
    $prompt = [$prompt]
}
```

Reply:
@JasonTheAdams Handled here https://github.com/WordPress/php-ai-client/pull/55/commits/b9c4aa696f6fd16b66f7f8ca397c7571e9d69c0d